x-pack/plugin/rollup/src/main/java/org/elasticsearch/xpack/rollup/v2/TransportRollupAction.java

```diff
@@ -277,126 +398,126 @@ private static XContentBuilder getDynamicTemplates(XContentBuilder builder) thro
             .endArray();
     }
 
-    /**
-     * Creates the rollup mapping properties from the provided {@link RollupActionConfig}.
-     */
-    private static XContentBuilder getProperties(XContentBuilder builder, RollupActionConfig config) throws IOException {
-        builder.startObject("properties");
-
-        RollupActionGroupConfig groupConfig = config.getGroupConfig();
-        RollupActionDateHistogramGroupConfig dateHistogramConfig = groupConfig.getDateHistogram();
-        String dateField = dateHistogramConfig.getField();
-        String dateIntervalType = dateHistogramConfig.getIntervalTypeName();
-        String dateInterval = dateHistogramConfig.getInterval().toString();
-        String tz = dateHistogramConfig.getTimeZone() != null
-            ? dateHistogramConfig.getTimeZone()
-            : RollupActionDateHistogramGroupConfig.DEFAULT_TIMEZONE;
+    private void manageDataStream(String sourceIndexName, String rollupIndexName, ActionListener<AcknowledgedResponse> listener) {
+        // Update cluster state for the data stream to add the rollup index and remove the source index
+        submitUnbatchedTask("update-rollup-metadata", new ClusterStateUpdateTask() {
+            @Override
+            public ClusterState execute(ClusterState currentState) {
+                Metadata metadata = currentState.metadata();
+                Metadata.Builder metadataBuilder = Metadata.builder(metadata);
+                IndexAbstraction sourceIndexAbstraction = metadata.getIndicesLookup().get(sourceIndexName);
+                // If rolling up a backing index of a data stream, replace the source index with
+                // the rolled up index to the data stream
+                if (sourceIndexAbstraction.getParentDataStream() != null) {
+                    DataStream originalDataStream = sourceIndexAbstraction.getParentDataStream().getDataStream();
+                    Index rollupIndex = metadata.index(rollupIndexName).getIndex();
+                    Index sourceIndex = metadata.index(sourceIndexName).getIndex();
+                    DataStream updatedDataStream = originalDataStream.replaceBackingIndex(sourceIndex, rollupIndex);
+                    metadataBuilder.put(updatedDataStream);
+                }
+                return ClusterState.builder(currentState).metadata(metadataBuilder.build()).build();
+            }
 
-        builder.startObject(dateField)
-            .field("type", DateFieldMapper.CONTENT_TYPE)
-            .startObject("meta")
-            .field(dateIntervalType, dateInterval)
-            .field(RollupActionDateHistogramGroupConfig.CalendarInterval.TIME_ZONE, tz)
-            .endObject()
-            .endObject();
+            @Override
+            public void clusterStateProcessed(ClusterState oldState, ClusterState newState) {
+                // 7. Delete the source index
+                deleteSourceIndex(sourceIndexName, rollupIndexName, listener);
```

Comment:
Will we always want to delete the source index? I get that we do if we're replacing the source in the data stream but if it isn't in a data stream is that required? Maybe I'm thinking about this funny - maybe we have to because of index patterns or something.

Reply:
Do not deleting the source index when an index is not a data stream backing index has certain benefits (mainly flexibility), but also adds complexity in certain areas such as ILM.

We have discussed this topic in the past and we have agreed that the souce index should be deleted in any case, even if it's not a data stream index.